Source/ACE.Entity/AceCharacter.cs

```diff
@@ -20,20 +20,22 @@ public AceCharacter(uint id)
 
             // Required default properties for character login
             // FIXME(ddevec): Should we have constants for (some of) these things?
-            SetIntProperty(PropertyInt.ItemType, (uint)ItemType.Creature);
             AceObjectDescriptionFlags = (uint)(ObjectDescriptionFlag.Stuck | ObjectDescriptionFlag.Player | ObjectDescriptionFlag.Attackable);
             WeenieHeaderFlags = (uint)(WeenieHeaderFlag.ItemsCapacity | WeenieHeaderFlag.ContainersCapacity | WeenieHeaderFlag.Usable | WeenieHeaderFlag.RadarBehavior);
+            WeenieClassId = 1;
+            SetDataIdProperty(PropertyDataId.Icon, 100667446);
+            SetIntProperty(PropertyInt.ItemType, (uint)Enum.ItemType.Creature);
             SetIntProperty(PropertyInt.ItemsCapacity, 102);
             SetIntProperty(PropertyInt.ContainersCapacity, 7);
-            WeenieClassId = 1;
-            Deleted = false;
-            DeleteTime = 0;
+            SetBoolProperty(PropertyBool.IsDeleted, false);
```

Comment:
These should operate on properties directly as opposed to calling the setproperty function.